### ISSUE TITLE
bug(oxlint@0.0.20) oxlint reports incorrect error for rule `eslint-plugin-react(jsx-key)`

### ISSUE THREAD

**Issue by andreasmcdermott** [oxc-project/oxc#1679]
With the following React code:

```
function App() {
  return (
    <div className="App">
      {[1, 2, 3, 4, 5].map((val) => {
        const text = <strong>{val}</strong>;
        return <button key={val}>{text}</button>;
      })}
    </div>
  );
}
```

`oxlint` gives the following error:

```
eslint-plugin-react(jsx-key): Missing "key" prop for element in iterator.
   ╭─[src/App.tsx:3:1]
 3 │     <div className="App">
 4 │       {[1, 2, 3, 4, 5].map((val) => {
   ·                        ─┬─
   ·                         ╰── Iterator starts here
 5 │         const text = <strong>{val}</strong>;
   ·                       ───┬──
   ·                          ╰── Element generated here
 6 │         return <button key={val}>{text}</button>;
   ╰────
  help: Add a "key" prop to the element in the iterator (https://react.dev/learn/rendering-lists#keeping-list-items-in-order-with-key).
```

This is incorrect, since `strong` is rendered within the `button`.
This error is not displayed if the strong is added directly into the `button`.

Here is a [reproduction on Code Sandbox](https://codesandbox.io/p/devbox/dazzling-breeze-92ly29?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clq5lim9a00083b6h14mk4zjp%2522%252C%2522sizes%2522%253A%255B60%252C40%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clq5lim9a00023b6hla5ty0yk%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clq5lim9a00053b6hy77w9bm7%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clq5lim9a00073b6h464zphxd%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B100%252C0%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clq5lim9a00023b6hla5ty0yk%2522%253A%257B%2522id%2522%253A%2522clq5lim9a00023b6hla5ty0yk%2522%252C%2522tabs%2522%253A%255B%255D%257D%252C%2522clq5lim9a00073b6h464zphxd%2522%253A%257B%2522id%2522%253A%2522clq5lim9a00073b6h464zphxd%2522%252C%2522tabs%2522%253A%255B%255D%257D%252C%2522clq5lim9a00053b6hy77w9bm7%2522%253A%257B%2522id%2522%253A%2522clq5lim9a00053b6hy77w9bm7%2522%252C%2522activeTabId%2522%253A%2522clq5lweaq00373b6eow1shi3w%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clq5lweaq00373b6eow1shi3w%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TERMINAL%2522%252C%2522shellId%2522%253A%2522clq5lwef500g5edipc3aebdl5%2522%257D%255D%257D%257D%252C%2522showDevtools%2522%253Afalse%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)

**Comment by camc314** [comment]
Ooo this is interesting. thank you for reporting and for the detail in your report!

i will fix

**Comment by camc314** [comment]
Closed by #1686
